pygmt/datatypes/grid.py

```diff
@@ -5,6 +5,8 @@
 import ctypes as ctp
 from typing import ClassVar
 
+import numpy as np
+
 # Constants for lengths of grid header variables.
 #
 # Note: Ideally we should be able to get these constants from the GMT shared library
```

Comment:
@weiji14 I feel you were right. We should put the `_GMT_GRID_HEADER` and some common constants/functions into `pygmt/datatypes/header.py` instead. Should we do the move now?

Reply:
Yeah, move the `_GMT_GRID_HEADER` class wrapped in #3127 and the parsing functions here to `pygmt/datatypes/header.py`.

Reply:
OK, done in ab001441e490e8d1f035d89bb08a0df4158fdb2f.